circe/src/main/scala/org/http4s/circe/CirceInstances.scala

```diff
@@ -25,11 +25,15 @@ trait CirceInstances {
     jsonEncoderWithPrinter(defaultPrinter)
 
   def jsonEncoderWithPrinter(printer: Printer): EntityEncoder[Json] =
-    EntityEncoder[String].contramap[Json] { json =>
-      // Comment from ArgonautInstances (which this code is based on):
-      // TODO naive implementation materializes to a String.
-      // See https://github.com/non/jawn/issues/6#issuecomment-65018736
-      printer.pretty(json)
+    EntityEncoder[Array[Byte]].contramap[Json] { json =>
+      val bytes = printer.prettyByteBuffer(json)
+      if (bytes.remaining < 1) {
```

Comment:
There is a `ByteVectorChunk <: Chunk[Byte]` in http4s, which doesn't do a great deal of good in fs2-0.9, because it has a slow concatAll.  `ByteVectorChunk` is in fs2-master, and I expect it to give us a performance boost.  I think the best reason to use `ByteVectorChunk` now is as a marker for things that should be the real `ByteVectorChunk` when it's available.